### ISSUE TITLE
Field width 536870912 is too long

### ISSUE THREAD

**Issue by BladeMF** [phpactor/phpactor#2144]
I keep getting this error for some reason:
`Fatal error: Field width 536870912 is too long in /root/.shared_extensions/phpactor3/lib/WorseReflection/Core/Type/StringLiteralType.php on line 34`

the code there is this:
```php
        if ($right instanceof StringLiteralType) {
            return new self(sprintf('%s%s', $this->value, (string)$right->value()));
        }
```

**Comment by BladeMF** [comment]
This saved me:
```php
			return new self(sprintf(
				'%s%s',
				substr($this->value, 0, 1024),
				substr((string)$right->value(), 1024)
			));
```

**Comment by BladeMF** [comment]
Still not sure what caused it.

**Comment by dantleech** [comment]
Yes that should definitely have been truncated, there us little to no value in storing the contents if the string

**Comment by dantleech** [comment]
would you be willing to create a PR?

**Comment by BladeMF** [comment]
Sure, but I am not sure what the length should be. I am a bit overwhelmed at the moment - I've been putting off my PR about switches for enabling and disabling different language server services for a very long time now, but I hope in a couple of weeks I would be able to.

**Comment by dantleech** [comment]
no worries, but basically the length can be arbitrary, e.g. 255 chars, maybe adding `**TRUNCATED**` if it's truncated

**Comment by BladeMF** [comment]
Sounds good.

**Comment by dantleech** [comment]
should be fixed as of #2200
